test/index.test.js

```diff
@@ -28,6 +28,37 @@ test('[useCallback] sinon-style', function(assert) {
   });
 });
 
+test('[useCallback] sinon-style onCall', function(assert) {
+  var getObject = AWS.stub('S3', 'getObject');
+  getObject.onCall(0).yields(null, data);
```

Comment:
I think `onCall` has existed, but it didn't allow you to change the response. From their [docs](http://sinonjs.org/releases/v2.3.7/stubs/):

> As of Sinon version 1.8, you can use the onCall method to make a stub respond differently on consecutive calls.
>
> Note that in Sinon version 1.5 to version 1.7, multiple calls to the yields* and callsArg* family of methods define a sequence of behaviors for consecutive calls. As of 1.8, this functionality has been removed in favor of the onCall API.

So I guess we could have used `yields` here instead of upgrading sinon?